### PR TITLE
feat: cancel buy now from listing

### PR DESCRIPTION
**Pull request by chandni-vault-hill**
## Description

https://github.com/Vault-Hill/vhc-marketplace-app/issues/43

### CODE REVIEW

**Review by jmulq — CHANGES_REQUESTED** [review]

File: src/services/queries.ts [hunk 1]
```diff
@@ -62,6 +62,9 @@ export const ASSET_LIST_FETCHER = ASSET_FETCHER.activeListing(LISTING_FETCHER)
 export const ASSET_LISTING_QUERY = query$.asset(ASSET_LIST_FETCHER)
 
 export const CANCEL_BID_MUTATION = mutation$.cancelBid(LISTING_FETCHER)
+
+export const ASSET_DATA_FETCHER = asset$.tokenId.tokenAddress
```
[the remark this answers is not in the document]
Comment (chandni-vault-hill) [reply src/services/queries.ts:66]: Yes to cancel the buy now, need two inputs, will need this

File: src/features/partials/Popups/CancelBuyNow.tsx [hunk 2]
```diff
@@ -1,20 +1,44 @@
...
 
 const CancelBuyNow = () => {
   const dispatch = useDispatch()
+  const [cancelBuyNowMutation, { isLoading }] = useCancelBuyNowListingMutation()
+  const [buyNow] = useAtom(cancelBuyNowAtom)
+  const [errorMessage, setErrorMessage] = useState(false)
+
   const cancel = async () => {
-    dispatch(closeModal())
+    const { tokenId, tokenAddress } = buyNow as Asset
```
[the remark this answers is not in the document]
Comment (chandni-vault-hill) [reply src/features/partials/Popups/CancelBuyNow.tsx:19]: Yes it was complaining for undefined, but will keep a check

File: src/features/partials/Popups/SellAssetSubmitted.tsx [hunk 3]
```diff
@@ -1,7 +1,22 @@
...
+import { useNavigate, useParams } from 'react-router-dom'
 import { Modal } from '../../shared/Modal'
+import { Button, ButtonColors, ButtonSizes } from '../../shared/Button'
+import { closeModal } from '../../../state/popup.slice'
+import { useAppDispatch } from '../../../state'
 
 const SellAssetSubmitted = () => {
+  const navigate = useNavigate()
+  const dispatch = useAppDispatch()
+  const params: any = useParams() as { tokenId: string }
```
[the remark this answers is not in the document]
Comment (chandni-vault-hill) [reply src/features/partials/Popups/SellAssetSubmitted.tsx:11]: Fixed

File: src/features/partials/AssetDetails/index.tsx [hunk 4]
```diff
@@ -34,18 +38,15 @@ const AssetDetails = ({ asset }: { asset: AssetWithListing | undefined }) => {
     }
   }
 
-  const handleSellClick = () => {
-    setListing(asset?.activeListing)
+  const handleSaleClick = () => {
```
Comment (jmulq) [inline src/features/partials/AssetDetails/index.tsx:41]: Just split this into two functions. Sale isn't being clicked when we unlist. I would just have `handleSell` and `handleUnlist` or something similar

File: src/features/partials/Popups/SellAssetSubmitted.tsx [hunk 5]
```diff
@@ -1,7 +1,22 @@
...
+import { useNavigate, useParams } from 'react-router-dom'
 import { Modal } from '../../shared/Modal'
+import { Button, ButtonColors, ButtonSizes } from '../../shared/Button'
+import { closeModal } from '../../../state/popup.slice'
+import { useAppDispatch } from '../../../state'
 
 const SellAssetSubmitted = () => {
+  const navigate = useNavigate()
+  const dispatch = useAppDispatch()
+  const params: { tokenId: string } = useParams() as { tokenId: string }
```
Comment (jmulq) [inline src/features/partials/Popups/SellAssetSubmitted.tsx:11]: you don't need to cast this. `useParams<{ tokenId: string }>()` should work.

**Review by jmulq — CHANGES_REQUESTED** [review]

File: src/features/partials/Popups/SellAssetSubmitted.tsx (hunk 5 above)
Comment (chandni-vault-hill) [reply src/features/partials/Popups/SellAssetSubmitted.tsx]: Fixed

File: src/features/partials/AssetDetails/index.tsx (hunk 4 above)
Comment (chandni-vault-hill) [reply src/features/partials/AssetDetails/index.tsx]: Separated the functions

**Review by jmulq — APPROVED** [review]

**Pull request merged**